### PR TITLE
Initial commit get

### PR DESCRIPTION
**Pull request by CharlesSquirel**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
# Code Review Report

## Task Goal
The goal of the task was to implement an endpoint for fetching JSON data. The data source should be `trains.json`.

## Requirements
- `trains.json` should be fetched as data source.
- Required endpoint schema: `GET /trains`

## Code Changes Review

The code changes seem to cover the task requirements. The endpoint `GET /trains` has been implemented and it fetches data from `trains.json`. 

However, there is a minor bug in the error handling section of the code. In the `catch` block, the error object is referred to as `err` but when logging the error to the console, `error` is used instead. This will cause a ReferenceError if an error occurs during JSON parsing.

Here is the problematic section:

```javascript
catch(err) {
    console.error(error);
    res.status(500).send('Wystąpił błąd podczas parsowania danych JSON.');
}
```

It should be corrected to:

```javascript
catch(err) {
    console.error(err);
    res.status(500).send('Wystąpił błąd podczas parsowania danych JSON.');
}
```

Other than this minor bug, the code changes seem to be in line with the task requirements.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
The task goal was to implement an endpoint for fetching JSON data.

### Requirements
- `trains.json` should be fetched as a data source.
- Required endpoint schema: `GET /trains`.

### Code changes review

The provided code changes seem to cover the task requirements. The developer has implemented a `GET /trains` endpoint in the `index.js` file. This endpoint reads the `trains.json` file, parses the JSON data, and sends it as a response.

Here are some points to note:

1. The developer correctly uses the `fs.readFile` function to read the `trains.json` file. The path to the file (`data/trains.json`) is also correct assuming that the `data` directory is at the same level as `index.js`.

2. If an error occurs while reading the file, the server responds with a 500 status code and a message indicating that an error occurred during file reading. This is good practice for error handling.

3. The developer correctly parses the JSON data using `JSON.parse`. If an error occurs during parsing, the server responds with a 500 status code and a message indicating that an error occurred during JSON parsing. This is also good practice for error handling.

4. The developer correctly sets the 'Content-Type' header to 'application/json' before sending the response.

5. The developer correctly formats the JSON data using `JSON.stringify` with an indentation of 2 spaces. This makes the JSON data easier to read.

6. The developer correctly sends the formatted JSON data as a response using `res.send`.

7. The developer correctly updates the port number from 4000 to 3000.

In conclusion, the provided code changes cover the task requirements and there are no apparent bugs in the code.
